timesketch/ui/graphs/graph-view.component.html

```diff
@@ -0,0 +1,54 @@
+<div *ngIf="state.type == 'empty'">
+    <h2>Please type a query.</h2>
+</div>
+
+<div *ngIf="state.type == 'loading'">
+  <h2>Loading...</h2>
+</div>
+
+<div *ngIf="state.type == 'ready'">
+  <div class="graph-view__dropdown">
+    <style type="text/css">
+      input ~ ts-graphs-cytoscape-settings {
+        display: none;
+      }
+      input:checked ~ ts-graphs-cytoscape-settings {
+        display: block;
+      }
+    </style>
+    <input type="checkbox" id="toggleCytoscapeSettings" style="display: none">
+    <label for="toggleCytoscapeSettings"> Cytoscape settings </label>
+    <ts-graphs-cytoscape-settings [settings]="settings">
+    </ts-graphs-cytoscape-settings>
+  </div>
+
+  <ts-graphs-cytoscape
+    [elements]="state.elements"
+    [style]="style"
+    [layout]="layout"
+
+    [minZoom]="settings.minZoom"
```

Comment:
At some point, it might make sense to pass in a whole settings object rather than all sections individually.

Reply:
Then the change detection will not detect the changes to individual fields.

Reply:
There are multiple ways to do this. Either you can use immtuables for this or override ngDoCheck (https://angular.io/guide/lifecycle-hooks#lifecycle-sequence). It's also ok to leave as-is for now, since this is a bigger change.